pkg/actuators/machine/reconciler.go

```diff
@@ -342,6 +351,87 @@ func (r *Reconciler) updateLoadBalancers(instance *ec2.Instance) error {
 	return nil
 }
 
+// deletePlacementGroup deletes the placement group for the machine
+func (r *Reconciler) deletePlacementGroup() error {
+	placementGroupName := r.providerSpec.Placement.GroupName
+	if placementGroupName == "" {
+		// Nothing to do if the group name is empty
+		return nil
+	}
+
+	// Check that the placement group exists
+	placementGroups, err := r.awsClient.DescribePlacementGroups(&ec2.DescribePlacementGroupsInput{
+		GroupNames: []*string{aws.String(placementGroupName)},
+	})
+	if err != nil && !isAWS4xxError(err) {
+		// Ignore a 400 error as AWS will report an unknown placement group as a 400.
+		return fmt.Errorf("could not describe placement groups: %v", err)
+	}
+
+	if len(placementGroups.PlacementGroups) == 0 {
+		// This is the normal path, the named placement group doesn't exist
+		return nil
+	}
+	if len(placementGroups.PlacementGroups) > 1 {
+		return fmt.Errorf("expected 1 placement group for name %q, got %d", placementGroupName, len(placementGroups.PlacementGroups))
+	}
```

Comment:
I think at this point we should check that the placement group is tagged correctly, if the instance doesn't have the cluster ID tag, then we shouldn't remove it (the lack of this tag implies it wasn't created by MAPI)